cmd/otp/otp_servehttp_test.go

```diff
@@ -0,0 +1,199 @@
+				Expect(logCapture.Contains("stored SSH key in OTP map")).To(BeTrue(), "Log should indicate success of SSH key placement in globalMap")
+			})
+
+			DescribeTable("valid-but-borderline SSH keys - stores the SSH key in OTP map when given a valid key",
+				func(inputFn func() string, label string) {
+					input := inputFn()
+					GinkgoWriter.Printf("Running test for: %s\n", label)
+					GinkgoWriter.Printf("Valid key value is: %s\n", input)
+
+					req := httptest.NewRequest("POST", "/store", strings.NewReader(input))
+					testStorekey.ServeHTTP(rr, req)
+
+					Expect(rr.Code).To(Equal(http.StatusOK))
+					Expect(rr.Body.String()).NotTo(BeEmpty())
+					Expect(logCapture.Contains("stored SSH key in OTP map")).To(BeTrue(), "Log should indicate success of SSH key placement in globalMap")
+				},
+
+				Entry("key with trailing whitespace", func() string { return rsaKey + "\n\t" }, "rsa key with whitespace at the end"),
+				Entry("key with comment", func() string { return rsaKey + " user@example.com" }, "rsa key with comment at the end"),
+				Entry("very long key", func() string { return rsaKey + rsaKey + rsaKey }, "a very long rsa key"),
+				Entry("ed25519 key", func() string { return edKey }, "an ed25519 key"),
+			)
+		})
+
+		// Skipping while KFLUXINFRA-1569 in SSH key validation is being fixed
+		PContext("with invalid ssh key (storekey)", func() {
+			badPrefixKey := "ssh-xyz123 AAAAB3NzaC1yc2EAAAADAQABAAABAQCe72TIPRwW/tDWPbNdJg3a6Rqy1/9kM002NLCx82fAN8GTvUj6VAD4Nl9om5BT7o0tfCcYgpDmTDrl7QPhBGd5ew0VKHO8o++SwhG6QI2mR+867qIdXP1B1ZdxO8eYndIn+ssOZcmbp4XxrI5/xWSNAU2XMckuSeUFZRTDNUVoKYmDMgnZL+BV6eQKO4jJmtuctDku4yb7YjcJYw7L+LOU7fBnsEpzvPJ/P9pGckpVI5nYIdQIUBxmVloa6BiKCGbu4yzPZ2zWISPODyvv6cmKk3s+YGild/TrVC+aRRN4TpiYq5NVT0lkkHpzhShVPukOR4xXoqZQLWYwAPLuUIBn"
+			DescribeTable("returns 500 for invalid SSH keys (storekey)",
+				func(input string, label string) {
+					GinkgoWriter.Printf("Running test for: %s\n", label)
+					GinkgoWriter.Printf("Invalid key value is: `%s`\n", input)
+
+					req := httptest.NewRequest("POST", "/store", strings.NewReader(input))
+
+					testStorekey.ServeHTTP(rr, req)
+					Expect(rr.Code).To(Equal(http.StatusInternalServerError))
+					Expect(logCapture.Contains("failed to read request body")).To(BeTrue(), "Log should indicate a problem with the request's body")
+				},
+
+				Entry("empty string", "", "An empty string"),
+				Entry("non-ssh string", "not-an-SSH-key", "A non-ssh key string"),
+				Entry("bad prefix", badPrefixKey, "An ssh key string with a badly formed prefix"),
+				Entry("missing payload", "ssh-rsa AAAAB3Nz", "An ssh key string with truncated content"),
+			)
+		})
+	})
+})
+
+// Generates a real SSH public key string, with option to change the encryption algorithm. Can be broadened to more
+// algorithms such as diffie-hellman, if we ever choose to test it.
+func generateValidSSHKey(keyType string) (string, error) {
+	switch keyType {
+	case "rsa":
```

Comment:
I'd suggest to have a some constants here

```go
const (
	rsaKeyType	string = "rsa"
	ed25519KeyType	string = "ed25519"
)
```

```suggestion
	case rsaKeyType:
```

Reply:
Not sure what the benefits of it are. 
It's not like we're going to change the value of these strings, and we want to change the code in only one place and have it effect multiple places. 
Also, these are very specific algorithm names - they're not a timeout value or a value that might change in the future without also changing it's name. 
If we decide to mostly test ed25519 instead of rsa and switch ed25519 with diffie-hellman, we'll also have to change  <name>KeyType.

Reply:
OK. Not a hill I'm willing to die on. See latest commit.